docker-entrypoint.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -e
+
+__config="
+database:
+  host: $DATABASE_HOST
+  port: $DATABASE_PORT
+  user: $DATABASE_USER
+  driver: $DATABASE_DRIVER
+  password: $DATABASE_PASSWORD
+  database: $DATABASE_NAME
+
+token: $DISCORD_TOKEN"
+
+if [ -n "$DEV_GUILD" ]; then
+  __config="
+$__config
+dev_guild: $DEV_GUILD"
+fi
+
+echo "$__config" > /usr/local/app/docker.config.yml
+
+echo -n "Waiting for database.."
+while ! nc -z $DATABASE_HOST $DATABASE_PORT 2>/dev/null; do
+    echo -n "."
+    sleep 1
+done
+
+
+echo -e \\n"Database ready"
+
+masonite-orm migrate -C database/config.py -d database/migrations || echo "Migration failed but continuing..."
+
+exec "$@"
```

Comment:
`sh` suffices here after a few modifications

```suggestion
#!/bin/sh
set -e

__config=$(cat <<EOF
database:
  host: $DATABASE_HOST
  port: $DATABASE_PORT
  user: $DATABASE_USER
  driver: $DATABASE_DRIVER
  password: $DATABASE_PASSWORD
  database: $DATABASE_NAME

token: $DISCORD_TOKEN
EOF
)

if [ -n "$DEV_GUILD" ]; then
  __config="$__config
dev_guild: $DEV_GUILD"
fi

echo "$__config" > /usr/local/app/docker.config.yml

echo -n "Waiting for database.."
while ! nc -z $DATABASE_HOST $DATABASE_PORT 2>/dev/null; do
    echo -n "."
    sleep 1
done

echo -e "\nDatabase ready"

masonite-orm migrate -C database/config.py -d database/migrations || echo "Migration failed but continuing..."

exec "$@"

```

Reply:
Is it going to work without `cat` ing the DEV_GUILD as well? In general, this seems to be a lot of tinkering for not a lot of gain (getting rid of one, arguably small dependency).

Reply:
In fact, `sh` works perfectly without any modifications. The `cat` thing is more of a security thing than anything else. In certain environments, indentation, line jumps, etc. can cause problems, but for `DEV_GUILD` there is none of that, and there's no need for it.

*But even the first `cat` isn't needed: as I said, it works with docker. It may not work in other environments, but it's a *docker entrypoint*, so the environment should always be the same. So you can get away with just changing `#!/bin/bash` to `#!/bin/sh`.

Reply:
Gotcha. I have changed the shell to `/bin/sh` and removed bash.